dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -11,74 +13,73 @@ This can cause confusion, for example, when a function might only work on strict
 
 
-## Use of "Topological"
+## `H3_EXPORT` and C name collisions
 
-We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+To avoid C name collisions, we should build the bindings (`h3-py`, `h3-java`, `h3-go`, and `h3-node`) with a standard function prefix, using the `H3_EXPORT` C macro.
+The `h3-js` binding would not need this.
 
-See, for example: https://github.com/uber/h3-js/issues/53
+The proposed prefix is `h3_`.
 
+## Functions
 
-## General Function Names
+### Conventions for "property getters" and transforms
 
-|          Current name         |                     Proposed name                      |
-|-------------------------------|--------------------------------------------------------|
-| *Does Not Exist (DNE)*        | `isValidIndex`                                         |
-| `h3IsValid`                   | `isValidCell`                                          |
-| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge`                                  |
-| `h3IsPentagon`                | `isPentagon`                                           |
-| `h3IsResClassIII`             | `isResClassIII`                                        |
-| `h3IndexesAreNeighbors`       | `areNeighborCells`                                     |
-| `h3ToParent`                  | `getParent` or `getCellParent` or `cellToParent`       |
-| `h3ToChildren`                | `getChildren` or `getCellChildren` or `cellToChildren` |
-| `numHexagons`                 | `getNumCells`                                          |
-| `getRes0Indexes`              | `getRes0Cells`                                         |
-| `getPentagonIndexes`          | `getPentagons`                                         |
-| `h3GetBaseCell`               | `getBaseCellNumber`                                    |
-| `h3GetResolution`             | `getResolution` or `getCellResolution`                 |
-| `geoToH3`                     | `geoToCell`                                            |
-| `h3ToGeo`                     | `cellToGeo`                                            |
-| `h3ToGeoBoundary`             | `cellToGeoBoundary`                                    |
-| `compact`                     |                                                        |
-| `uncompact`                   |                                                        |
-| `polyfill`                    |                                                        |
+- use `get` prefix for
+    + constant data (`getNumCells`, `getRes0Cells`)
+    + object properties (`getResolution`, `getBaseCellNumber`)
+- use `to` to denote transforms
+    + different representations of the same object
+    + when doing a lossy transformation to a new object (`cellToParent`, `pointToCell`)
+- do not use `get` or `two` for *computations*
```

Comment:
typo: two -> to

Reply:
He's technically right, you should not use `two` as a prefix to computations ;)

Reply:
haha, thanks

Reply:
fixed.